src/inatinqperf/adaptors/milvus_adaptor.py

```diff
@@ -0,0 +1,164 @@
+"""Milvus adaptor."""
+
+from collections.abc import Sequence
+from enum import Enum
+
+from datasets import Dataset as HuggingFaceDataset
+from loguru import logger
+from pymilvus import (
+    DataType,
+    MilvusClient,
+    connections,
+    utility,
+)
+from tqdm import tqdm
+
+from inatinqperf.adaptors.base import DataPoint, Query, SearchResult, VectorDatabase
+from inatinqperf.adaptors.metric import Metric
+
+
+class MilvusIndexType(str, Enum):
+    """Enum for various index types supported by Milvus. For more details, see https://milvus.io/docs/index.md?tab=floating."""
+
+    IVF_FLAT = "IVF_FLAT"
+    IVF_SQ8 = "IVF_SQ8"
+    IVF_PQ = "IVF_PQ"
+    HNSW = "HNSW"
+    HNSW_SQ = "HNSW_SQ"
+    HNSW_PQ = "HNSW_PQ"
+
+    @classmethod
+    def _missing_(cls, value: str) -> "MilvusIndexType | None":
+        value = value.lower()
+        for member in cls:
+            if member.value.lower() == value:
+                return member
+        return None
+
+
+class Milvus(VectorDatabase):
+    """Milvus vector database."""
+
+    DATABASE_NAME: str = "default"
+    COLLECTION_NAME: str = "collection_name"
+    INDEX_NAME: str = f"{COLLECTION_NAME}_index"
```

Comment:
I feel like all of these should be killed and used as keyword arguments in the constructor instead.